x/staking/types/msg.go

```diff
@@ -182,12 +182,17 @@ func (msg MsgEditValidator) ValidateBasic() sdk.Error {
 		return sdk.NewError(DefaultCodespace, CodeInvalidInput, "transaction must include some information to modify")
 	}
 
+	if msg.CommissionRate != nil {
+		if msg.CommissionRate.GT(sdk.OneDec()) || msg.CommissionRate.LT(sdk.ZeroDec()) {
+			return sdk.NewError(DefaultCodespace, CodeInvalidInput, "commission rate must be between 0 and 1, inclusive")
+		}
+	}
+
 	return nil
 }
 
-//______________________________________________________________________
-
 // MsgDelegate - struct for bonding transactions
+// TODO: Why do we need to store the denomination in `Value`
```

Comment:
because you may be attempting to delegate different types of tokens - I suppose you could just send in an integer and then deduct those tokens if the account has the relevant staking denom. However - this could cause errors down the road for message senders (accidentally delegating to the wrong chain, but has both types of tokens) - so including the denom here just enforces that the msg is intentional... so I think we should keep it

Reply:
we rely in many places on exactly one token - `BondDenom` - being the staking token

Reply:
sure, I secede don't really mind, would make the code base cleaner...  we could have more client side checks to ensure users are performing their intended choices